Integrations/integration-ArcherRSA.yml

```diff
@@ -0,0 +1,626 @@
+           '             <sessionToken>' + token + '</sessionToken>' +
+           '             <moduleId>' + appId + '</moduleId>' +
+           '             <fieldValues> ' +
+           '                 <![CDATA[ ' +
+           '                    <Record>' +
+                                   fields +
+           '                    </Record>' +
+           '                ]]>' +
+           '            </fieldValues>' +
+           '      </CreateRecord>' +
+           '   </soap:Body> ' +
+           '</soap:Envelope>'
+    }
+
+    function updateContentSoapRequest(token, appId, contentId, fields) {
+        return '<?xml version="1.0" encoding="utf-8"?> ' +
+        '<soap:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchemainstance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">' +
+        ' <soap:Body>' +
+        ' <UpdateRecord xmlns="http://archer-tech.com/webservices/">' +
+        ' <sessionToken>' + token + '</sessionToken>' +
+        ' <moduleId>' + appId + '</moduleId>' +
+        ' <contentId>' + contentId +'</contentId>' +
+        ' <fieldValues>' +
+        '<![CDATA[' +
+            '<UpdateRecord>' +
+                fields +
+            '</UpdateRecord>' +
+        ']]>' +
+        '</fieldValues>' +
+        ' </UpdateRecord>' +
+        ' </soap:Body>' +
+        ' </soap:Envelope>'
+    }
+
+    function getLevelsByApplicationId(token) {
+
+        var lvlResponse = sendRestRequest('levelIdByModuleId', '', token, true);
+        var levels = dq(lvlResponse, 'RequestedObject.Id');
+        console.log('---Levels: ' + levels);
+        return levels;
+    }
+
+    function createMappingFieldIdToName(levels, token) {
+        console.log("in createMappingFieldIdToName");
+        var mapping = {};
+        if(!levels) {
+            throw 'Mapping levels to Fields failed, levels = ' + JSON.stringify(levels);
+        }
+        if(!Array.isArray(levels)) {
+            levels = [levels];
+        }
+        levels.forEach(function (lvl) {
+            args.levelId = lvl;
+            var fieldsLevelInfo = sendRestRequest('fieldByLevelId','',token, true);
+            console.log("fieldsLevelInfo:\n" + JSON.stringify(fieldsLevelInfo));
+            if (!Array.isArray(fieldsLevelInfo)) {
+               fieldsLevelInfo = [fieldsLevelInfo];
+            }
+            fieldsLevelInfo.forEach(function (field) {
+                mapping[field.RequestedObject.Id] = {Type: field.RequestedObject.Type, Name: field.RequestedObject.Name, levelId: lvl};
+            })
+        });
+        console.log("Mapping:\n" + JSON.stringify(mapping));
+        return mapping;
+    }
+
+    function createMappingNameToFieldId(levels, token) {
+        console.log("in createMappingFieldIdToName");
+        var mapping = {};
+        if(!levels) {
+            throw 'Mapping levels to Fields failed, levels = ' + JSON.stringify(levels);
```

Comment:
level does not exits (can be null) and you are doing JSON.stringify on it.. should be instead:
`throw 'Mapping levels to Fields failed, levels not provided`